h/api/search/config.py

```diff
@@ -0,0 +1,334 @@
+            'filter': ['icu_folding']
+        }
+    }
+}
+
+DOCUMENT_MAPPING = {
+    '_id': {'path': 'id'},
+    '_source': {'excludes': ['id']},
+    'analyzer': 'keyword',
+    'date_detection': False,
+    'properties': {
+        'id': {'type': 'string', 'index': 'no'},
+        'annotator_schema_version': {'type': 'string'},
+        'created': {'type': 'date'},
+        'updated': {'type': 'date'},
+        'title': {'type': 'string', 'analyzer': 'standard'},
+        'link': {
+            'type': 'nested',
+            'properties': {
+                'type': {'type': 'string'},
+                'href': {'type': 'string'},
+            }
+        },
+        'dc': {
+            'type': 'nested',
+            'properties': {
+                # by default elastic search will try to parse this as
+                # a date but unfortunately the data that is in the wild
+                # may not be parsable by ES which throws an exception
+                'date': {'type': 'string'}
+            }
+        }
+    }
+}
+
+DOCUMENT_ANALYSIS = {}
+
+
+def configure_index(client, index=None):
+    """Configure the elasticsearch index."""
+
+    if index is None:
+        index = client.index
+
+    # Ensure the ICU analysis plugin is installed
+    _ensure_icu_plugin(client.conn)
+
+    # Construct desired mappings and analysis settings
+    mappings = {}
+    analysis = {}
+    _append_config(mappings,
+                   analysis,
+                   client.t.annotation,
+                   ANNOTATION_MAPPING,
+                   ANNOTATION_ANALYSIS)
+    _append_config(mappings,
+                   analysis,
+                   client.t.document,
+                   DOCUMENT_MAPPING,
+                   DOCUMENT_ANALYSIS)
+
+    # Try to create the index with the correct settings. This will not fail if
+    # the index already exists.
+    _create_index(client.conn, index, {
+        'mappings': mappings,
+        'settings': {'analysis': analysis},
+    })
+
+    # For indices we didn't just create: try and update the analysis and
+    # mappings to their current values. May throw an exception if elasticsearch
+    # throws a MergeMappingException indicating that the index cannot be
+    # updated without reindexing.
+    _update_index_analysis(client.conn, index, analysis)
+    _update_index_mappings(client.conn, index, mappings)
+
+
+def _ensure_icu_plugin(conn):
+    """Ensure that the ICU analysis plugin is installed for ES."""
+    # Pylint issue #258: https://bitbucket.org/logilab/pylint/issue/258
+    #
+    # pylint: disable=unexpected-keyword-arg
+    names = [x.strip() for x in conn.cat.plugins(h='component').split('\n')]
+    if 'analysis-icu' not in names:
+        message = ("ICU Analysis plugin is not installed for Elasticsearch\n"
+                   "  See the installation instructions for more details:\n"
+                   "  https://github.com/hypothesis/h/blob/master/"
+                   "INSTALL.rst#installing")
```

Comment:
This is indeed out of date. I've updated it with a link directly to the elasticsearch-icu repo, as our install document now just says the same thing -- that you must have it installed.